source/2_platform/rex_directx/src/dxgi/adapter.cpp

```diff
@@ -0,0 +1,130 @@
+#include "rex_directx/dxgi/adapter.h"
+
+#include "rex_directx/dxgi/util.h"
+#include "rex_directx/log.h"
+#include "rex_std/stdlib.h"
+
+namespace
+{
+  const uint32 g_adapter_description_size = 128;
+
+  //-------------------------------------------------------------------------
+  /**
+   * The VendorId is a unique identifier assigned by the PCI-SIG (Peripheral Component Interconnect Special Interest Group)
+   * to identify the manufacturer of a PCI device, including graphics adapters.
+   * The VendorId is a 16-bit unsigned integer that is typically included in
+   * the PCI Configuration space of the device.
+   */
+  enum class Vendors
+  {
+    UNKNOWN = 0,
+    AMD     = 0x1002,
+    NVIDIA  = 0x10DE,
+    INTEL   = 0x163C
+  };
+
+  //-------------------------------------------------------------------------
+  rsl::string vendor_to_string(s32 vendor)
+  {
+    switch(static_cast<Vendors>(vendor))
+    {
+      case Vendors::AMD: return rsl::string("AMD");
+      case Vendors::NVIDIA: return rsl::string("NVIDIA");
+      case Vendors::INTEL: return rsl::string("INTEL");
+      default: return rsl::string("Unknown Vendor");
+    }
+  }
+
+  ////-------------------------------------------------------------------------
+  // rsl::string to_multibyte(const tchar* wideCharacterBuffer, count_t size)
```

Comment:
shouldn't this get removed?

Reply:
Depends are you going to put it into rsl or not? 
Cuz that's the reason why it is there

Reply:
the C standard has support for this afaik wctomb?

Reply:
Yes and we don't cover some of those functions in RSL?

Reply:
I use wcstombs_s as we speak